spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindingServiceConfiguration.java

```diff
@@ -245,6 +246,28 @@ public MessageChannel resolveDestination(String name) throws DestinationResoluti
 
 			});
 		}
+
+		@Bean
+		public static BeanPostProcessor messageHandlerHeaderPropagationBeanPostProcessor(
+				final SpringIntegrationProperties springIntegrationProperties) {
+			return new BeanPostProcessor() {
+				@Override
+				public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+					// TODO: Filter out beans created by SCSt (not currently necessary)
+					if (bean instanceof AbstractReplyProducingMessageHandler) {
```

Comment:
Not sure if that is critical right now or will be possible according your TODO, but the bean can be proxy coming to this BPP, therefore won't be `instanceof` for class. Only interfaces are possible here.
Or something like this:
```
Class<?> targetClass = bean.getClass();
if (AopUtils.isAopProxy(bean) {
targetClass = AopUtils.getTargetClass(this.handler);

}
boolean replyMessageHandler = AbstractReplyProducingMessageHandler.class.isAssignableFrom(targetClass);
```